recipe/meta.yaml

```diff
@@ -109,43 +133,25 @@ outputs:
     {% set tests_to_skip = tests_to_skip + " or test_full_reimport" %}          # [ppc64le or s390x]
     {% set tests_to_skip = tests_to_skip + " or test_pep338" %}                 # [ppc64le or s390x]
     {% set tests_to_skip = tests_to_skip + " or test_no_typing_extensions" %}   # [ppc64le or s390x]
-    # 2022/5/5: E       AssertionError: (131, 1.54987431981169249551435343964035e-09, 4.0035173453529620614007210953362e-19, 'arcsinh')
-    # Flawed test when using MKL
-    # https://github.com/numpy/numpy/issues/16769
-    {% set tests_to_skip = tests_to_skip + " or test_overrides" %}  # [blas_impl == 'mkl']
-    # https://github.com/numpy/numpy/issues/15243
-    {% set tests_to_skip = tests_to_skip + " or test_loss_of_precision" %}  # [s390x or ppc64le]
-    # It should be fixed by https://github.com/numpy/numpy/issues/20426
-    # but it still fails on some platforms.
-    {% set tests_to_skip = tests_to_skip + " or test_new_policy" %}   # [s390x or ppc64le or (osx and arm64)]
-    # On osx-arm64: FAILED core/tests/test_limited_api.py::test_limited_api - subprocess.CalledProcessor
-    {% set tests_to_skip = tests_to_skip + " or test_limited_api" %}   # [osx and arm64]
-    # ?
-    {% set tests_to_skip = tests_to_skip + " or test_gcd_overflow" %}  # [s390x]
-    # ?
-    {% set tests_to_skip = tests_to_skip + " or test_herm_cases" %}  # [osx and arm64]
-    {% set tests_to_skip = tests_to_skip + " or test_generalized_herm_cases" %}  # [osx and arm64]
-    # ?
-    {% set tests_to_skip = tests_to_skip + " or test_partial_iteration_cleanup" %}  # [osx or win64]
 
     test:
       requires:
         - pip     # force installation or `test_api_importable` will fail
+        - cython >=0.29.30,<3.0
+        - wheel
         - setuptools <60.0.0
-        - pytest
         - hypothesis
-        - {{ compiler('c') }}  # [not osx]
-        - {{ compiler('cxx') }}  # [not osx]
-        - {{ compiler('fortran') }}  # [not osx]
-        - nomkl  # [x86 and blas_impl != 'mkl']
+        - pytest
+        - pytz
+        - pytest-cov
+        - cffi  # [py<310]
+        - mypy
+        - typing_extensions
       commands:
```

Comment:
These new `test` requirements seem to be over specified. It's not clear why we would need `pytz`, `mypy`, `wheel`, etc. for tests, especially if we are not running the upstream test suite.

Reply:
The core packages for testing are only these:
```
  - pytest
  - pytest-cov
  - pytest-xdist
  - hypothesis
```

see https://github.com/numpy/numpy/blob/21cacafbeeff812cf49ca0102d1dd582b6d77e2b/environment.yml#L17-L20
I guess compilers are needed too

Reply:
There were additional specifications needed for the tests to run properly, for example typing-extensions was needed for some of the tests.